subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -320,7 +322,8 @@ public void visitInputs(InputVisitor visitor) {
                         : InputPropertyType.NON_INCREMENTAL;
                 String propertyName = inputFileProperty.getPropertyName();
                 visitor.visitInputFileProperty(propertyName, type, NON_IDENTITY, value, () -> {
-                    FileCollectionFingerprinter fingerprinter = fingerprinterRegistry.getFingerprinter(inputFileProperty.getNormalizer());
+                    FileCollectionFingerprinterSpec fingerprinterSpec = DefaultFileCollectionFingerprinterSpec.from(inputFileProperty.getNormalizer(), inputFileProperty.getDirectorySensitivity());
```

Comment:
I wonder if we need the indirection of creating a `DefaultFileCollectionFingerprinterSpec` or if we should just pass in the `InputFilePropertySpec`.

Perhaps we could allow `InputFilePropertySpec` to implement `FileNormalizationSpec`.

Reply:
Good idea of making `InputFilePropertySpec` implement `FingerprinterSpec` (I assume that's what you meant by `FileNormalizationSpec`).